### PR TITLE
Move tag key and device uuid length validation into SBVR rules

### PR DESCRIPTION
**Pull request by Page-**
Update pinejs from 14.10.0 to 14.10.1

Change-type: minor

### CODE REVIEW

File: src/migrations/00040-validation-checks.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
+ALTER TABLE "release tag"
```
Comment (Page-) [inline src/migrations/00040-validation-checks.sql:1]: I'll be looking into making this more automatic as well

File: src/migrations/00040-validation-checks.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	-- It is necessary that each application tag has a tag key that has a Length (Type) that is greater than 0.
+	ADD CCONSTRAINT "application tag$zPAVMu9ZY2npomham40YGgXx5N6Hau03dIo6x9gf6/E=" CHECK (0 < LENGTH("tag key")
+	AND LENGTH("tag key") IS NOT NULL
+	AND "tag key" = "tag key"
+	AND "tag key" IS NOT NULL)
+;
+
+ALTER TABLE "device"
+	-- It is necessary that each device has a uuid that has a Length (Type) that is equal to 32 or is equal to 62.
+	ADD CCONSTRAINT "device$NEO6SIKl2TSRYZL9mbgDqA4Nue1p1KopGLDoLhqbYfM=" CHECK ((LENGTH("uuid") = 32
```
Comment (xginn8) [inline src/migrations/00040-validation-checks.sql:19]:
```suggestion
	ADD CONSTRAINT "device$NEO6SIKl2TSRYZL9mbgDqA4Nue1p1KopGLDoLhqbYfM=" CHECK ((LENGTH("uuid") = 32
```

File: src/migrations/00040-validation-checks.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	OR LENGTH("uuid") = 62)
+	AND LENGTH("uuid") IS NOT NULL
+	AND "uuid" IS NOT NULL)
+;
+
+ALTER TABLE "device tag"
+	-- It is necessary that each device tag has a tag key that has a Length (Type) that is greater than 0.
+	ADD CCONSTRAINT "device tag$30aEY0OcDs3I/zbRIyNPL9K/I7WY+4PabIF1sxOvXKg=" CHECK (0 < LENGTH("tag key")
```
Comment (xginn8) [inline src/migrations/00040-validation-checks.sql:27]:
```suggestion
	ADD CONSTRAINT "device tag$30aEY0OcDs3I/zbRIyNPL9K/I7WY+4PabIF1sxOvXKg=" CHECK (0 < LENGTH("tag key")
```

File: src/migrations/00040-validation-checks.sql [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	-- It is necessary that each release tag has a tag key that has a Length (Type) that is greater than 0.
+	ADD CONSTRAINT "release tag$vGZu47lKJepQVH+hgSZNuUPdet2cG96akz3Yc8hta3A=" CHECK (0 < LENGTH("tag key")
+	AND LENGTH("tag key") IS NOT NULL
+	AND "tag key" = "tag key"
+	AND "tag key" IS NOT NULL)
+;
+
+ALTER TABLE "application tag"
+	-- It is necessary that each application tag has a tag key that has a Length (Type) that is greater than 0.
+	ADD CCONSTRAINT "application tag$zPAVMu9ZY2npomham40YGgXx5N6Hau03dIo6x9gf6/E=" CHECK (0 < LENGTH("tag key")
```
Comment (xginn8) [inline src/migrations/00040-validation-checks.sql:11]:
```suggestion
	ADD CONSTRAINT "application tag$zPAVMu9ZY2npomham40YGgXx5N6Hau03dIo6x9gf6/E=" CHECK (0 < LENGTH("tag key")
```

**Review by thgreasi — COMMENTED** [review]
Assuming that this still returns 400 and fixing the typos in the migration, absolutely loved this.
(Based on the consistency of the typos, I'm guessing that you are already have a ~working migration generator, right?)

File: src/migrations/00041-validation-checks.sql [hunk 5]
```diff
@@ -0,0 +1,23 @@
+ALTER TABLE "release tag"
+	-- It is necessary that each release tag has a tag key that has a Length (Type) that is greater than 0.
+	ADD CONSTRAINT "release tag$vGZu47lKJepQVH+hgSZNuUPdet2cG96akz3Yc8hta3A=" CHECK (0 < LENGTH("tag key")
+	AND LENGTH("tag key") IS NOT NULL
+	AND "tag key" = "tag key"
```
Comment (thgreasi) [inline src/migrations/00041-validation-checks.sql:5]: I guess that this isn't wrong but it's weird seeing it here anyway.

Comment (Page-) [reply src/migrations/00041-validation-checks.sql:5]: Yeah that should be easy enough to opimize out, not sure offhand why it's even included tbh

**Review by thgreasi — APPROVED** [review]
:tada:

**Pull request merged**